### PR TITLE
IGNITE-12344 Remote listener of IgniteMessaging has to run with appropriate SecurityContext.

### PR DESCRIPTION
**Pull request by dgarus**
Remote listener of IgniteMessaging has to run with appropriate SecurityContext.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteMessagingImpl.java [hunk 1]
```diff
@@ -207,7 +210,7 @@ private void send0(@Nullable Object topic, Collection<?> msgs, boolean async) th
         guard();
 
         try {
-            GridContinuousHandler hnd = new GridMessageListenHandler(topic, (IgniteBiPredicate<UUID, Object>)p);
+            GridContinuousHandler hnd = new GridMessageListenHandler(topic, securityAwareBiPredicate(p));
```
Comment (NSAmelchev) [inline modules/core/src/main/java/org/apache/ignite/internal/IgniteMessagingImpl.java]: Do we should add predicate to async implementation `remoteListenAsync` ?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/security/messaging/MessagingRemoteSecurityContextCheckTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteMessaging;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.internal.IgniteInterruptedCheckedException;
+import org.apache.ignite.internal.processors.security.AbstractRemoteSecurityContextCheckTest;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.junit.Test;
+
+/**
+ *
```
Comment (NSAmelchev) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/security/messaging/MessagingRemoteSecurityContextCheckTest.java]: Add description, please

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteMessagingImpl.java (hunk 1 above)
Comment (dgarus) [reply modules/core/src/main/java/org/apache/ignite/internal/IgniteMessagingImpl.java]: yes!

File: modules/core/src/test/java/org/apache/ignite/internal/processors/security/messaging/MessagingRemoteSecurityContextCheckTest.java (hunk 2 above)
Comment (dgarus) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/security/messaging/MessagingRemoteSecurityContextCheckTest.java]: ok

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteFeatures.java [hunk 3]
```diff
@@ -88,7 +88,10 @@
     CONT_QRY_SECURITY_AWARE(21),
 
     /** Long operations dump timeout. */
-    LONG_OPERATIONS_DUMP_TIMEOUT(30);
+    LONG_OPERATIONS_DUMP_TIMEOUT(30),
+
+    /** ContinuousQuery with security subject id support. */
+    IGNITE_MESSAGING_SECURITY_AWARE(31);
```
Comment (ascherbakoff) [inline modules/core/src/main/java/org/apache/ignite/internal/IgniteFeatures.java]: I think a new feature is not necessary, you can use CONT_QRY_SECURITY_AWARE. 
Moreover, that's the point in using features at all ?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/security/messaging/MessagingRemoteSecurityContextCheckTest.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+import org.apache.ignite.IgniteMessaging;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.internal.IgniteInterruptedCheckedException;
+import org.apache.ignite.internal.processors.security.AbstractRemoteSecurityContextCheckTest;
+import org.apache.ignite.lang.IgniteBiPredicate;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.junit.Test;
+
+/**
+ * Testing operation security context when the remoute listener of IgniteMessaging is executed on remote nodes.
```
Comment (ascherbakoff) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/security/messaging/MessagingRemoteSecurityContextCheckTest.java]: Typo in "remoute".

Comment (dgarus) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/security/messaging/MessagingRemoteSecurityContextCheckTest.java]: fixed

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteFeatures.java (hunk 3 above)
Comment (dgarus) [reply modules/core/src/main/java/org/apache/ignite/internal/IgniteFeatures.java]: I think it's confused to use CONT_QRY_SECURITY_AWARE for rolling update the IgniteMessage security feature.
I've fixed JavaDoc.

Comment (dgarus) [reply modules/core/src/main/java/org/apache/ignite/internal/IgniteFeatures.java]: @ascherbakoff  I have deleted the IGNITE_MESSAGING_SECURITY_AWARE feature.

**Pull request closed without merging**